src/app/modules/submission/multirespondent-submission/multirespondent-submission.controller.ts

```diff
@@ -279,45 +285,70 @@ const _createSubmission = async ({
       meta: {
         ...logMeta,
         ...createReqMeta(req),
-        currentWorkflowStep: submissionContent.workflowStep,
+        currentWorkflowStep: currentStepNumber,
         formId: form._id,
         submissionId,
       },
       error: err,
     })
   }
+
+  if (isTest || form.admin.betaFlags.mrfEmailNotifications) {
```

Comment:
Same note here regarding the reference comment to allow quick lookup during cleanups